### PR TITLE
Api promotions hotels

### LINKED ISSUE #68: Create api crud for resource Promotion - Hotels
- [x] Read Promotion
- [x] Update Promotion
- [x] Create Promotion
- [x] Delete Promotion

### PR DESCRIPTION
**Pull request by paulotijero**
Resolve #68

### CODE REVIEW

File: app/controllers/api/promotions_hotels_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  def index
+    render json: Hotel.find(params[:hotel_id]).promotions
+  end
+  
+  def show
+    render json: @promotion
+  end
+  
+  def create
+    promotion = Promotion.new(promotion_params)
```
Comment (condef5) [inline app/controllers/api/promotions_hotels_controller.rb]: Hi :), I think you should create promotions through the hotel, so it is no longer necessary to pass as parameters in the `promotionable_type`,` promotionable_id` API, Something like this:

```ruby
@hotel = Hotel.find(params[:hotel_id])
@promotion = @hotel.promotions.new(hotel_params)
if @promotion.save
  render json: @promotion, status: :created
else
  render json: { errors: @promotion.errors}
end
```

File: app/controllers/api/promotions_hotels_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    render json: { message: e.message }, status: :not_found
+  end
+  
+  private    
+  def set_promotion 
+    @promotion = Promotion.find(params[:id])
+  end
+  
+  def promotion_params
+    params.permit(:start_date, :end_date, :discount_type, :discount_amount, :promotionable_type, :promotionable_id)
```
Comment (condef5) [inline app/controllers/api/promotions_hotels_controller.rb]: This should be enough:
```ruby
params.permit(:start_date, :end_date, :discount_type, :discount_amount)
```

File: app/serializers/promotion_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class PromotionSerializer < ActiveModel::Serializer
+  include Rails.application.routes.url_helpers
+  attributes :id, :start_date, :end_date, :discount_type, :discount_amount, :promotionable_type, :promotionable_id
```
Comment (condef5) [inline app/serializers/promotion_serializer.rb]: Hey, the attributes `:promotionable_type, :promotionable_id` are not necessary

File: app/controllers/api/promotions_hotels_controller.rb (hunk 1 above)
Comment (paulotijero) [reply app/controllers/api/promotions_hotels_controller.rb]: you're right, now I change it

File: spec/controllers/api/promotions_hotels_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      }
+      room_deleted = Room.where(
+        id: @promotion.id,
+        hotel_id: @hotel.id
+      )
+      expect(room_deleted.size).to eq(0)
+    end
+  end
+
+end
```
Comment (liannivin) [inline spec/controllers/api/promotions_hotels_controller_spec.rb]: Thanks for the comments. It is very helpful.

**Pull request merged**